tests/define_status_test_project.py

```diff
@@ -0,0 +1,64 @@
+import os
+
+import flow
+from flow import FlowProject
+
+
+class _TestProject(FlowProject):
+    pass
+
+
+group1 = _TestProject.make_group(name="group1")
+group2 = _TestProject.make_group(name="group2")
+
+
+@_TestProject.label
+def default_label(job):
+    return True
+
+
+@_TestProject.label
+def negative_default_label(job):
+    return False
+
+
+@_TestProject.label("named_label")
+def anonymous_label(job):
+    return True
+
+
+@_TestProject.label
+def b_is_even(job):
+    try:
+        return job.sp.b % 2 == 0
+    except AttributeError:
+        return False
+
+
+@flow.cmd
+@group1
+@_TestProject.pre(b_is_even)
+@_TestProject.post.isfile("world.txt")
+def op1(job):
+    return 'echo "hello" > {job.ws}/world.txt'
+
+
+@_TestProject.operation
+@group1
+@_TestProject.post.true("test")
+def op2(job):
+    job.document.test = os.getpid()
+
+
+@_TestProject.operation
+@group2
+@_TestProject.post.true("test3_true")
+@_TestProject.post.false("test3_false")
+@_TestProject.post.not_(lambda job: job.doc.test3_false)
```

Comment:
@b-butler If I recall correctly, the project that this was copy-pasted from included `not_` for the purposes of increasing test coverage. We had no other tests of that condition.

It is equivalent to the `post.false("test3_false")` condition but will not raise an error because it is the last postcondition. The earlier postconditions are not met until after the operation runs, so the "early exit" on an unmet postcondition prevents this from running (and raising). However, this is not great design.

For this PR, we can simplify the conditions. I don't think status information will be affected if we were to simplify to the following:

```python
@_TestProject.operation
@group2
@_TestProject.post.true("test2")
def op3(job):
    job.document.test2 = True
```

Separately, we could change the implementation of the `not_` condition in the test file this came from so that the potential error is eliminated:

```python
@_TestProject.post.not_(lambda job: job.doc.get("test3_false", None))
```